### PR TITLE
Generate valid_signals list dynamically

### PR DESCRIPTION
**Pull request by sammyshj**
This PR is in continuation of #110

### CODE REVIEW

**Comment by sammyshj** [comment]
@meejah The code now handles only TorProtocolError.
The test seems to be failing with pypy. Can you help me out with that?

File: txtorcon/torcontrolprotocol.py [hunk 1]
```diff
@@ -716,13 +716,13 @@ def _bootstrap(self, *args):
...
-        # FIXME XXX there is now "GETINFO signal/names"
-        self.valid_signals = ["RELOAD", "DUMP", "DEBUG", "NEWNYM",
-                              "CLEARDNSCACHE"]
+        try:
+            self.valid_signals = yield self.get_info('signal/names')
+            self.valid_signals = self.valid_signals['signal/names']
+        except TorProtocolError:
+            self.valid_signals = ["RELOAD", "DUMP", "DEBUG", "NEWNYM",
+                                  "CLEARDNSCACHE"]
+            raise RuntimeError("Tor version does not support signal/names")
```
Comment (meejah) [inline txtorcon/torcontrolprotocol.py]: This isn't an error; we've figured out the valid_signals list (i.e. using the static list, not the one from getinfo).

**Comment by sammyshj** [comment]
@meejah I have removed the RuntimeError from the except block. If the code looks good then should I squash the commits?

**Comment by meejah** [comment]
Yes, please squash

**Comment by sammyshj** [comment]
@meejah This can be merged now :)

**Comment by meejah** [comment]
great, thanks!

**Pull request merged**